osbrain/agent.py

```diff
@@ -1759,15 +1759,15 @@ def close(self, alias):
 
         sock.close(linger=get_linger())
 
-    def close_all(self):
+    def close_all(self, linger=get_linger()):
         """
         Close all non-internal zmq sockets.
         """
         # Each socket might be pointed by different keys
         sockets_to_delete = []
         for sock in self._get_unique_external_zmq_sockets():
             sockets_to_delete.append(sock)
-            sock.close(linger=get_linger())
+            sock.close(linger)
```

Comment:
This is not the expected behavior.

Do you know when that `get_linger()` gets executed?

In my opinion we should leave it as it was before and skip if Windows fails, then create a new issue to investigate this issue and/or a workaround (i.e.: serializing the global configuration to child processes?). But for now it is easier if we simply skip.

Reply:
Why is it not the expected behavior? I don't understand what you mean. The value of linger can be passed as an optional parameter, and if it isn't, `get_linger()` is called, as it was before. How does this differ from the expected behavior?

Reply:
```python
conf = dict(linger=1)


def get_linger():
    return conf['linger']


def default_conf(linger=get_linger()):
    return linger


if __name__ == '__main__':
    print(default_conf())
    conf['linger'] = 2
    print(default_conf())
```

Reply:
Oh. I didn't know that happened. What about something like this:
```python
conf = dict(linger=1)


def get_linger():
    return conf['linger']


def default_conf(linger=None):
    if linger is None:
        linger = get_linger()
    return linger


if __name__ == '__main__':
    print(default_conf())
    conf['linger'] = 2
    print(default_conf())
```

Reply:
Yeah, although that should be in a separate PR. Basically you'd be adding support for passing a linger parameter to .close_all(). That would require new tests. If you want to implement this we should also probably add the same parameter to .close() (with tests again).

As it is a non-trivial change, I would not add it here. Let us for now keep this PR for only making AppVeyor succeed (even if skipping some tests for now).

Reply:
Ok. I will skip the linger tests as well, then.